packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/model/inheritance/single-discriminator/src/Generated/Models/Eagle.cs

```diff
@@ -11,7 +11,7 @@ public partial class Eagle : Bird
     {
         public Eagle(int wingspan) : base("eagle", wingspan) => throw null;
 
-        internal Eagle(IList<Bird> friends, IDictionary<string, Bird> hate, Bird partner, int wingspan, IDictionary<string, BinaryData> additionalBinaryDataProperties) : base("eagle", wingspan, additionalBinaryDataProperties) => throw null;
+        internal Eagle(IList<Bird> friends, IDictionary<string, Bird> hate, Bird partner, string kind, int wingspan, IDictionary<string, BinaryData> additionalBinaryDataProperties) : base(kind, wingspan, additionalBinaryDataProperties) => throw null;
```

Comment:
I discovered an issue with the previous implementation of this. Since the discriminator value isn't exposed as a parameter in the serialization ctor for a model, there can be problems while deserialization / serializing the model. ie. a payload can have the discriminator property and if we try to deserialize it into the model, since the model's serialization ctor doesn't accept it as a param, if the serialization format is != W then that property gets stored as raw data. If we try to serialize the same model we will serialize the discriminator property as it was set by default during model instantiation and also the binary data, which would have the same duplicate property name (`kind` in this case).

As a solution, I've included the discriminator as a parameter in the serialization ctor. This follows what autorest [does today](https://github.com/Azure/autorest.csharp/blob/63e709c86843c86c51eadcfa87f4b9b26f3d3d47/test/CadlRanchProjectsNonAzure/type/model/inheritance/single-discriminator/src/Generated/Models/Eagle.cs#L38), and it seems like it was part of the initial [proposed design](https://github.com/microsoft/typespec/issues/3500) so I'm not sure if that was just missed/overlooked.

The existing tests didn't previously fail since for discriminators we weren't handling binary/raw data, but that fix is included in this PR.